File: app/src/main/java/com/sagar/android/logutil/MainActivity.java
package com.sagar.android.logutil;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        ((ApplicationClass)getApplicationContext()).getLogUtil().logD("Test debug log.");
        ((ApplicationClass)getApplicationContext()).getLogUtil().logD("CustomDebugTag","Test log with ");
    }
}

File: app/src/main/java/com/sagar/android/logutil/MainActivity.kt
package com.sagar.android.logutil

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.sagar.android.logutilmaster.LogLevel
import com.sagar.android.logutilmaster.LogUtilContract

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        (applicationContext as ApplicationClass).logUtil.registerForCallback(
                object : LogUtilContract {
                    override fun logged(message: String, logLevel: LogLevel, logTag: String) {
                        Log.v("log_util_call_back", ">> $message >> $logTag")
                    }
                }
        )

        (applicationContext as ApplicationClass).logUtil.logD("Test debug log.")
        (applicationContext as ApplicationClass).logUtil.logD("CustomDebugTag", "Test log with ")
    }
}
